clang/test/SemaSYCL/intel-fpga-local.cpp

```diff
@@ -182,7 +182,7 @@ void check_ast()
   // No diagnostic is emitted because the arguments match.
   // Duplicate attribute is silently ignored.
   //CHECK: VarDecl{{.*}}var_bankwidth 'int'
-  //CHECK: IntelFPGAMemoryAttr{{.*}}Implicit memory Default
+  //CHECK: IntelFPGAMemoryAttr{{.*}}Implicit
```

Comment:
What causes this difference?

Reply:
Thanks @premanandrao for the reviews.

The change is happening due to the removal of Attribute spelling `` [[intelfpga:: memory]]``.

Before this PR: AST was generated like: //CHECK: IntelFPGAMemoryAttr{{.*}}Implicit memory Default
Now AST looks like this : //CHECK: IntelFPGAMemoryAttr{{.*}}Implicit Default

Reply:
Can you please add the `Default` back into the test? As it stands, it looks like printing the argument broke otherwise.

Reply:
>>Can you please add the Default back into the test? 

Yes, I will add this. Thank you.

>>As it stands, it looks like printing the argument broke otherwise.

Agreed

Reply:
It still looks like I am missing something here.  Is the "memory" attr not gneerated for the intel::fpga_memory spelling?

Reply:
When doing a textual dump for an attribute, the `TextNodeDumper` calls into code generated by ClangAttrEmitter.cpp and that generated code will dump out the spelling used when there are multiple possible spellings. Because we removed a spelling, the generated code no longer needs to dump out which spelling was used.

Reply:
Thanks, TIL!

Reply:
>>Can you please add the Default back into the test?

I have added the Default back into the test.

Reply:
CodeGen part for memory attribute:
```
void CodeGenModule::generateIntelFPGAAnnotation(
    const Decl *D, llvm::SmallString<256> &AnnotStr) {
  llvm::raw_svector_ostream Out(AnnotStr);
  if (D->hasAttr<IntelFPGARegisterAttr>())
    Out << "{register:1}";
  if (auto const *MA = D->getAttr<IntelFPGAMemoryAttr>()) {
    IntelFPGAMemoryAttr::MemoryKind Kind = MA->getKind();
    Out << "{memory:";
    switch (Kind) {
    case IntelFPGAMemoryAttr::MLAB:
    case IntelFPGAMemoryAttr::BlockRAM:
      Out << IntelFPGAMemoryAttr::ConvertMemoryKindToStr(Kind);
      break;
    case IntelFPGAMemoryAttr::Default:
      Out << "DEFAULT";
      break;
    }
```

Here is Sema code that generates the implicit attribute on AST:
```
// If the declaration does not have an [[intel::fpga_memory]]
  // attribute, this creates one as an implicit attribute.
  if (!D->hasAttr<IntelFPGAMemoryAttr>())
    D->addAttr(IntelFPGAMemoryAttr::CreateImplicit(
        Context, IntelFPGAMemoryAttr::Default));
```

Reply:
> When doing a textual dump for an attribute, the `TextNodeDumper` calls into code generated by ClangAttrEmitter.cpp and that generated code will dump out the spelling used when there are multiple possible spellings. Because we removed a spelling, the generated code no longer needs to dump out which spelling was used.

Yes. This same behavior is happening with bank_bits attribute that generates implicit Numbank attribute.

```
//CHECK: VarDecl{{.*}} bb_reg 'unsigned int [4]'
  //CHECK: IntelFPGANumBanksAttr{{.*}}Implicit
  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 4
  //CHECK: IntelFPGAMemoryAttr{{.*}}Implicit Default
  //CHECK: IntelFPGABankBitsAttr
  //CHECK-NEXT: ConstantExpr{{.*}}'int'
  //CHECK-NEXT: value: Int 2
  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 2
  //CHECK-NEXT: ConstantExpr{{.*}}'int'
  //CHECK-NEXT: value: Int 3
  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 3
  //expected-error@+2 1{{'fpga_register' and 'bank_bits' attributes are not compatible}}
  [[intel::bank_bits(2, 3)]]
  [[intel::fpga_register]]
  //expected-note@-2 1{{conflicting attribute is here}}``